plugin.program.newgrounds/addon.py

```diff
@@ -48,64 +49,67 @@ def build_url(query):
     'Connection': 'keep-alive'
 }
 
-def get_request(url, data=None):
-    cookie_jar.load(cookie_file, ignore_discard=True, ignore_expires=True)
-    opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(cookie_jar))
-    urllib2.install_opener(opener)
-    # Debug
+def getFeatured(content, mode, cType):
+    url = "https://newgrounds.app/featured.cache.php"
+    data = None
+
+    req = urllib2.Request(url, data, hdr)
+    response = urllib2.urlopen(req)
+    r = json.loads(response.read())
+    #xbmcgui.Dialog().ok('DEBUG', json.dumps(r))
+    for fm in r[content]:
+        #xbmcgui.Dialog().ok('DEBUG', fm['content_url'])
+        url = build_url({'mode': mode, 'contentID': fm['content_url']})
+        li = xbmcgui.ListItem(fm['title'], iconImage=fm['thumbnail_newgrounds_url'] + fm['thumbnail'])
+        li.setInfo(cType, {'title': fm['title']})
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+    xbmcplugin.endOfDirectory(addon_handle)
+
+def searchContent(content, mode, cType):
+    if (int(args['page'][0]) > 1): #next page
+        search = args['search_term'][0]
+    else: #initial search
+        kb = xbmc.Keyboard('', 'search')
+        kb.doModal()
+        if kb.isConfirmed():
```

Comment:
what if the user cancelled? Then the `search` variable does not exist and stuff breaks.